### DISCUSSION TITLE
Welcome to Factory Discussions!

### DISCUSSION THREAD

**Discussion by hmlongco** [hmlongco/Factory#59] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by herveperoteau** [comment]
**Migration Swift 6.2 with Factory 3.0** 

I m currently working on app where I try to evaluate the effort to migrate to Swift6.2

To start this migration, I used Factory develop branch to have 3.0
All is good and run well.

But if I activate in my Package.swift the setting .defaultIsolation(MainActor.self), I have problems.

I have problems to call Container autoRegister (Call to main actor-isolated static method 'autoRegister' in a synchronous nonisolated context)... so I tried to embed it inside a Task @MainActor, it compiled well but it crash inside a UseCase using @Injected(\.xxxx).  _Task 23: Fatal error: DeleteSynchronizedDocumentsUseCaseProtocol was not registered_

**Do you have already find a good strategy about migration Swift 6.2 and factory ?**

Thank you for your recommandations.
Hervé

**Comment by hmlongco** [comment]
There are several issues at play here, including the one you encountered. At this point in time my only suggestion is to create your own `@MainActor func setup() { ... }` function on the container and call `Container.shared.setup()` during app initialization.
